drivers/counter/counter_nrfx_rtc.c

```diff
@@ -0,0 +1,287 @@
+{
+	nrfx_rtc_disable(&get_nrfx_config(dev)->rtc);
+
+	return 0;
+}
+
+static u32_t counter_nrfx_read(struct device *dev)
+{
+	return nrfx_rtc_counter_get(&get_nrfx_config(dev)->rtc);
+}
+
+static int counter_nrfx_set_alarm(struct device *dev,
+				  const struct counter_alarm_cfg *alarm_cfg)
+{
+	const struct counter_nrfx_config *nrfx_config = get_nrfx_config(dev);
+	const nrfx_rtc_t *rtc = &nrfx_config->rtc;
+	u32_t cc_val;
+
+	if (alarm_cfg->ticks > get_dev_data(dev)->wrap) {
+		return -EINVAL;
+	}
+
+	if (nrfx_config->alarm_cfgs[alarm_cfg->channel_id]) {
+		return -EBUSY;
+	}
+
+	cc_val = alarm_cfg->ticks + (alarm_cfg->absolute ?
+					0 : nrfx_rtc_counter_get(rtc));
+	cc_val = (cc_val > get_dev_data(dev)->wrap) ?
+			(cc_val - get_dev_data(dev)->wrap) : cc_val;
+	nrfx_config->alarm_cfgs[alarm_cfg->channel_id] = alarm_cfg;
+
+	nrfx_rtc_cc_set(rtc, ID_TO_CC(alarm_cfg->channel_id), cc_val, true);
+
+	return 0;
+}
+
+static void _disable(struct device *dev, u8_t id)
+{
+	const struct counter_nrfx_config *config = get_nrfx_config(dev);
+
+	nrfx_rtc_cc_disable(&config->rtc, ID_TO_CC(id));
+	config->alarm_cfgs[id] = NULL;
+}
+
+static int counter_nrfx_disable_alarm(struct device *dev,
+				     const struct counter_alarm_cfg *alarm_cfg)
+{
+	_disable(dev, alarm_cfg->channel_id);
+
+	return 0;
+}
+
+static int counter_nrfx_set_wrap_alarm(struct device *dev,
+				       u32_t ticks,
+				       counter_wrap_callback_t callback,
+				       void *user_data)
+{
+	const struct counter_nrfx_config *nrfx_config = get_nrfx_config(dev);
+	const nrfx_rtc_t *rtc = &nrfx_config->rtc;
+	struct counter_nrfx_data *dev_data = get_dev_data(dev);
+
+	for (int i = 0; i < counter_get_num_of_channels(dev); i++) {
+		/* Overflow can be changed only when all alarms are
+		 * disables.
+		 */
+		if (nrfx_config->alarm_cfgs[i]) {
+			return -EBUSY;
+		}
+	}
+
+	nrfx_rtc_cc_disable(rtc, WRAP_CH);
+	nrfx_rtc_counter_clear(rtc);
+
+	dev_data->wrap_cb = callback;
+	dev_data->wrap_user_data = user_data;
+	dev_data->wrap = ticks;
+	nrfx_rtc_cc_set(rtc, WRAP_CH, ticks, callback ? true : false);
+
+	return 0;
+}
+
+static u32_t counter_nrfx_get_pending_int(struct device *dev)
+{
+	return 0;
+}
+
+static void alarm_event_handler(struct device *dev, u32_t id)
+{
+	const struct counter_nrfx_config *config = get_nrfx_config(dev);
+	const struct counter_alarm_cfg *alarm_cfg = config->alarm_cfgs[id];
+	u32_t cc_val;
+
+	if (!alarm_cfg->handler) {
+		return;
+	}
+
+	cc_val = nrf_rtc_cc_get(config->rtc.p_reg, ID_TO_CC(id));
+
+	_disable(dev, id);
+	alarm_cfg->handler(dev, alarm_cfg, cc_val);
+}
+
+static void event_handler(nrfx_rtc_int_type_t int_type, void *p_context)
+{
+	struct device *dev = p_context;
+	struct counter_nrfx_data *data = get_dev_data(dev);
+
+	if (int_type == COUNTER_WRAP_INT) {
+		/* Manually reset counter when wrap is different than max wrap*/
+		if (data->wrap != COUNTER_MAX_WRAP) {
+			nrfx_rtc_counter_clear(&get_nrfx_config(dev)->rtc);
```

Comment:
yes, i think we could add PPI as an option. if this option is present in kconfig then user can learn that without this option counter will have accumulative error in period. In many cases it might not be an issue.